docs/source/models.rst

```diff
@@ -3655,23 +3655,113 @@ Evaluating with Extra Metrics
+
+The ``MetricValue`` class has three attributes:
+
+* ``scores``: used for per-row metrics 
+* ``aggregate_results``: a dictionary that maps metric names to their aggregate results
+* ``justification``: a per-row justification of the values in ``scores``
+The code block below demonstrates how to define a custom metric evaluation function:
+
+.. code-block:: python
+    from mlflow.metrics import MetricValue
+
+    def my_metric_eval_fn(predictions, targets, metrics):
+        scores = predictions
+        mymetric = np.sum(np.abs(predictions - targets))
+        return MetricValue(scores=scores, aggregate_results={"mymetric": mymetric})
+
+Once you have defined an ``eval_fn``, you then use ``make_metric()`` to wrap this ``eval_fn`` function into a metric.
+In addition to ``eval_fn``, ``make_metric()`` requires an additional parameter , ``greater_is_better``, for optimization purposes. This parameter
+indicates whether this is a metric we want to maximize or minimize.
+
+.. code-block:: python
+    from mlflow.metrics import make_metric
+
+    mymetric = make_metric(eval_fn=my_metric_eval_fn, greater_is_better=False)
+
+The extra metric allows you to either evaluate a model directly, or to evaluate an output dataframe. 
+
+To evaluate the model directly, you will have to provide ``mlflow.evaluate()`` either a pyfunc model
+instance, a URI referring to a pyfunc model, or a callable function that takes in the data as input 
+and outputs the predictions.
+
+.. code-block:: python
+    def model(x):
+        return x["inputs"]
+
+    eval_dataset = pd.DataFrame({"targets": [1,2,3,4,5,6,7,8], "inputs": [1,2,3,4,5,6,7,8]})
+
+    mlflow.evaluate(model, eval_dataset, targets="targets", extra_metrics=[mymetric])
+
+To directly evaluate an output dataframe, you can **omit** the ``model`` parameter. However, you will need
+ to set the ``predictions`` parameter in ``mlflow.evaluate()`` in order to evaluate an inference output dataframe. 
+
+.. code-block:: python
+
+    eval_dataset = pd.DataFrame({"targets": [1,2,3,4,5,6,7,8], "predictions": [1,2,3,4,5,6,7,8]})
+
+    result = mlflow.evaluate(data=eval_dataset, predictions="predictions", targets="targets", extra_metrics=[mymetric])
+
+When your model has multiple outputs, the model must return a pandas DataFrame with multiple columns. You must
+specify one column among the model output columns as the predictions column using the ``predictions`` parameter,
+and other output columns of the model will be accessible from the ``eval_fn`` via col_mapping. For example, if 
+your model has two outputs ``retrieved_context`` and ``answer``, you can specify ``answer`` as the predictions
+column, and ``retrieved_context`` column will be accessible from the ``eval_fn``:
+
+.. code-block:: python
+
+    def eval_fn(predictions, targets, metrics, retrieved_context):
+        return MetricValue(aggregate_results={"mymetric": np.sum(predictions == targets), "sum_of_retrieved_context": np.sum(retrieved_context)})
```

Comment:
IIRC, the metrics name is supposed to be specified in `make_metric`, and the key in aggregate_results would be the suffix. cc @sunishsheth2009 @annzhang-db @prithvikannan To give the recommendation.